babel/number_spelling.py

```diff
@@ -0,0 +1,549 @@
+                if digit.index == 0:
+                    if teen:
+                        digit.string = teens[digit.value]
+
+                if digit.index == 1:
+                    if teen:
+                        digit.string = ''
+                    elif digit.value > 0 and group.digits[0].value > 0:
+                        digit.suffix = '-'
+
+                # adding hundred suffix
+                if digit.index == 2 and digit.value > 0:
+                    digit.suffix = ' hundred'
+                    if group.digits[0].value > 0 or group.digits[1].value > 0:
+                        digit.suffix += ' and '
+
+                # run specialities for ordinals at last non-zero digit
+                if ordinal and side == number.integer and digit == number.last_nonzero:
+
+                    # add normal ordinals
+                    if digit.index < 2:
+                        digit.string = names_ord[digit.index][digit.value]
+                        number.suffix = ''
+
+                    if digit.index == 0:
+                        if teen:
+                            digit.string = teens_ord[digit.value]
+
+    fra_scale = ' ten hundred thousand'.split(' ')
+    # add suffix on fraction side
+    if number.fraction.value > 0:
+        number.fraction.suffix = ' ' + fra_scale[len(number.fraction)] + 'th'
+
+        if number.fraction.value > 1:
+            number.fraction.suffix += 's'
+
+
+@spell('hu_HU', ContextMaker(3, 3, 6))
+def hu_HU(number, ordinal):
+    """
+    Based on the official language guideline of Hungary:
+    http://hu.wikisource.org/wiki/A_magyar_helyes%C3%ADr%C3%A1s_szab%C3%A1lyai/Egy%C3%A9b_tudnival%C3%B3k#288.
+    http://hu.wikipedia.org/wiki/Wikip%C3%A9dia:Helyes%C3%ADr%C3%A1s/A_sz%C3%A1mok_%C3%ADr%C3%A1sa
+
+    Names of the powers of ten: http://hu.wikipedia.org/wiki/T%C3%ADz_hatv%C3%A1nyai
+
+    For doublechecking: http://helyesiras.mta.hu
+
+    Currently supports: +/- 10^66 - 10^-7
+
+    :copyright: (c) 2013 by Szabolcs Blága
+    :license: BSD, see LICENSE in babel for more details.
+    """
+
+    # name-parts for the Hungarian long scale (every 10^6 has an unique name)
+    scale_base = 'm b tr kvadr kvint szext szept okt non dec'
+    scale = ['', 'ezer'] + prod(scale_base, 'illió illiárd')
+    scale_ord = ['', 'ezre'] + prod(scale_base, 'illiomo illiárdo')
+    # alternate spelling (millió, ezermillió, billió, ezerbillió, stb.)
+
+    # avoid using decimal by comparing number of digits
+    if len(number.integer) > len(scale)*3:
+        number.string = 'leírhatatlanul ' + ('sokadik' if ordinal else 'sok') + ' :)'
+        return
+
+    if len(number.integer) < 2 and number.integer.value == 0:
+        number.integer.string = 'nulla'
+
+    # initialization running once
+    if number.fraction:
+        number.separator = ' egész '
+
+    if number.rounded:
+        number.prefix = 'kerekítve '
+
+    if number.negative:
+        number.prefix += 'mínusz '
+
+
+    names = (
```

Comment:
I'm unable to validate hungarian. This point is unrelated to the other discussion we had:

I'm of the opinion that if we implement spellers per locale, we'll want to have a directory for that and a file per locale so that locale specific and generic code is clearly seperated and locale specific spelling code can be easily looked up.

@erickwilder Any opinion from your side?